regular_expressions_intermediate/regular_expressions_intermediate.md

```diff
@@ -0,0 +1,540 @@
+
+The caret symbol `^` (indicating the beginning of a string) and dollar sign `$` (indicating the end of a string) are special characters called "anchors", which can be used when it's important to find patterns in a specific location.  
+
+<div class = "important">
+<b style="color: rgb(var(--color-highlight));">Important note</b><br>
+
+This is where one of your **flags** becomes important.  We're going to use Regex 101 with the "multiline" flag, which means that each line starts with a start-of-string and ends with an end-of-string.  As you try these yourself, make sure you're using a multiline flag as well!
+
+</div>
+
+### Start-of-string Anchoring
+
+Let's consider our mouse identifiers example (our wild type and PBMC humanized mice).  We want to find text entries that begin with subject identifiers, such as:  
+
+* PBMC-12345: listless, off feed
+* WT-234293 is unremarkable
+
+We don't, on the other hand, want to find texts that have mouse identifiers that aren't at the beginning of the line.  These should not be pattern matches:
+
+* the mouse PBMC-12345
+* subject WT-234293
+
+If we just [use the pattern for the identifier alone](https://regex101.com/r/oTOVnK/1), `(?:WT|PBMC)\-\d{5,6}`, we won't get that discrimination between line starters and non-line starters:
+
+![Using the pattern for the identifier alone results in matching the identifier when it starts a line as well as when it's in the middle of a line,](media/without_anchor.png)<!-- style = "border: 1px solid rgb(var(--color-highlight)); max-width:400px;" -->
+
+However, if [we add the `^` at the start of our pattern](https://regex101.com/r/4HxMlo/1), resulting in `^(?:WT|PBMC)\-\d{5,6}`, we can indicate that our pattern must immediately begin the string:
+
+![Using the pattern that has the anchor followed by the regex for the identifier results in matching the identifier when it starts a line but not matching the identifier when it's in the middle of a line.](media/with_anchor.png)<!-- style = "border: 1px solid rgb(var(--color-highlight)); max-width:400px;" -->
+
+<div class = "important">
+<b style="color: rgb(var(--color-highlight));">Important note</b><br>
+
+To correctly use `^` as the start-of-string token, you must not enclose it in square brackets.  If `^` appears inside square brackets, it will be intepreted as doing something quite different, as you are about to discover.
+
+</div>
+
+### End-of-string Anchoring 
+
+Let's check that strings **end** with one of our preapproved punctuation options.  We'd like to match problematic strings, which are ones that don't end in either a period, question mark, or exclamation point.  
+
+There's a helpful bit of grammar in regex that means "except for", when it appears as the first symbol within square brackets.  Interestingly enough, we use `^` to do this.  That symbol does double duty, because we also use `^` to indicate the start of a line, when it's not inside square brackets!  Placement is everything.  
```

Comment:
```suggestion
There's a helpful bit of grammar in regex that means "except for," when it appears as the first symbol within square brackets.  Interestingly enough, we use `^` to do this.  That symbol does double duty, because we also use `^` to indicate the start of a line, when it's not inside square brackets!  Placement is everything.  
```